core/src/main/java/ai/timefold/solver/core/impl/neighborhood/maybeapi/move/ChangeMoveDefinition.java

```diff
@@ -21,13 +23,12 @@ public ChangeMoveDefinition(PlanningVariableMetaModel<Solution_, Entity_, Value_
 
     @Override
     public MoveStream<Solution_> build(MoveStreamFactory<Solution_> moveStreamFactory) {
-        var enumeratingStream =
-                moveStreamFactory.forEachEntityValuePair(variableMetaModel)
-                        .filter((solutionView, entity, value) -> {
-                            Value_ currentValue = solutionView.getValue(variableMetaModel, Objects.requireNonNull(entity));
-                            return !Objects.equals(currentValue, value);
-                        });
-        return moveStreamFactory.pick(enumeratingStream)
+        var nodeSharingSupportFunctions =
+                ((DefaultMoveStreamFactory<Solution_>) moveStreamFactory).getNodeSharingSupportFunctions(variableMetaModel);
```

Comment:
Are the functions from `NodeSharingSupportFunctions` going to be reused? If not, we can add them to `ChangeMoveDefinition`, similar to how we integrated `ListChangeMoveDefinition::isValidChange?`.